File: server/PoultryFarmManager.Core/Models/Finance/Person.cs
namespace PoultryFarmManager.Core.Models.Finance;

public class Person : DbEntity
{
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string? Email { get; set; }
    public string? PhoneNumber { get; set; }
    public string? Location { get; set; }
}

File: client/webapp/src/types/person.ts
export interface Person {
    id: string;
    firstName: string;
    lastName: string;
    email?: string;
    phoneNumber?: string;
    location?: string;
}

export interface NewPerson {
    firstName: string;
    lastName: string;
    email?: string;
    phoneNumber?: string;
    location?: string;
}

export interface UpdatePerson {
    firstName?: string;
    lastName?: string;
    email?: string;
    phoneNumber?: string;
    location?: string;
}
